frame/contracts/src/storage/meter.rs

```diff
@@ -441,21 +444,15 @@ where
 		// contract execution does conclude and hence would lead to a double charge.
 		self.total_deposit = deposit.clone();
 
+		// We need to make sure that the contract's account exists.
+		T::Currency::transfer(origin, contract, ed, Preservation::Preserve)?;
+
+		System::<T>::inc_consumers(contract)?;
+
 		// Normally, deposit charges are deferred to be able to coalesce them with refunds.
 		// However, we need to charge immediately so that the account is created before
 		// charges possibly below the ed are collected and fail.
-		E::charge(
-			origin,
-			contract_info.deposit_account(),
-			&deposit.saturating_sub(&Deposit::Charge(ed)),
-			false,
-		)?;
-
-		System::<T>::inc_consumers(contract_info.deposit_account())?;
-
-		// We also need to make sure that the contract's account itself exists.
-		T::Currency::transfer(origin, contract, ed, Preservation::Preserve)?;
-		System::<T>::inc_consumers(contract)?;
+		E::charge(origin, contract, &deposit.saturating_sub(&Deposit::Charge(ed)), false)?;
```

Comment:
comment above seems off as well now, since you start by transferring ed

Reply:
Yeah, I don't think we need any comment at all there, maybe just the fact that we are excluding the `ed` from the `deposit` because it was transferred in the previous step. Would you add anything else?

Reply:
yep let's just remove it, also see my comment above